include/linker/devicetree_regions.h

```diff
@@ -7,6 +7,8 @@
  * Generate memory regions from devicetree nodes.
  */
 
+#define _DT_COMPATIBLE	zephyr_memory_region
```

Comment:
nit: internal helper macros should all be put under

```
/** @cond INTERNAL_HIDDEN */
...
/** @endcond */
```

Reply:
In that file there is already an `INTERNAL_HIDDEN` section, so I'm just moving the `_DT_COMPATIBLE` in there.